src/pymovements/stimuli/text_stimulus.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2023-2024 The pymovements Project Authors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Module for the TextDataFrame."""
+from __future__ import annotations
+
+import polars as pl
+
+import pymovements as pm  # pylint: disable=cyclic-import
+from pymovements.stimuli.stimulus import StimulusDataFrame
+
+
+class TextStimulusDataFrame(StimulusDataFrame):
+    def __init__(
+            self,
+            data: pl.DataFrame | None = None,
```

Comment:
Let's have the following signature for now, ignoring image data and focusing solely on AOIs. Let's also ignore the None case for now.

Also for the start, let's suppose we have character level AOIs just as the example file that we have.

The minimal contents of the AOIs should be:
- character: the character in the text string
- pixel_x: pixel x-position of the top left corner
- pixel_y: pixel y-position of the top left corner
- width: pixel width of aoi box
- height: pixel height of aoi box

It can potentially have additional columns
- page: the page index
- line: the line on the page
- column: the character index in the line, the naming can be improved here
- word: the word this character/substring belongs to

The `TextStimulus` constructor should then work similar to the `GazeDataFrame` constructor, where we rename columns to have consistent access later on.

I would propose this signature for the constructor:
```python
def __init__(
    self,
    aois: pl.DataFrame,
    *,
    character_column: str,
    pixel_x_column: str,
    pixel_y_column: str,
    width_column: str,
    height_column: str,
    page_column: str,
):
```